sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesResultCollection.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// Collection of <see cref="RecognizeCustomEntitiesActionResult"/> objects corresponding
+    /// to a batch of documents, and information about the batch operation.
+    /// </summary>
+    [DebuggerTypeProxy(typeof(RecognizeCustomEntitiesResultCollectionDebugView))]
+    public class RecognizeCustomEntitiesResultCollection : ReadOnlyCollection<RecognizeEntitiesResult>
+    {
+        internal RecognizeCustomEntitiesResultCollection(IList<RecognizeEntitiesResult> list, TextDocumentBatchStatistics statistics, string projectName, string deploymentName) : base(list)
+        {
+            Statistics = statistics;
+            DeploymentName = deploymentName;
+            ProjectName = projectName;
+        }
+
+        /// <summary>
+        /// Gets statistics about the documents and how it was processed
+        /// by the service.  This property will have a value when IncludeStatistics
+        /// is set to true in the client call.
+        /// </summary>
+        public TextDocumentBatchStatistics Statistics { get; }
+
+        /// <summary>
+        /// Gets the value of the property corresponding to the name of the
+        /// deployment for this action.
+        /// </summary>
+        public string DeploymentName { get; }
+
+        /// <summary>
+        /// Gets the value of the property corresponding to the name of the
+        /// project for this action.
+        /// </summary>
+        public string ProjectName { get; }
+
+        /// <summary>
+        /// Debugger Proxy class for <see cref="RecognizeCustomEntitiesResultCollectionDebugView"/>.
```

Comment:
nit: this is actually the debugger proxy class for `RecognizeCustomEntitiesResultCollection`.